homeassistant/components/hunterdouglas_powerview/config_flow.py

```diff
@@ -0,0 +1,116 @@
+"""Config flow for Hunter Douglas PowerView integration."""
+import logging
+
+from aiopvapi.helpers.aiorequest import AioRequest
+from aiopvapi.hub import Hub
+import async_timeout
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_HOST, CONF_NAME
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DOMAIN  # pylint:disable=unused-import
+from .const import HUB_EXCEPTIONS
+
+_LOGGER = logging.getLogger(__name__)
+
+DATA_SCHEMA = vol.Schema({vol.Required(CONF_HOST): str})
+HAP_SUFFIX = "._hap._tcp.local."
+
+
+async def validate_input(hass: core.HomeAssistant, data):
+    """Validate the user input allows us to connect.
+
+    Data has the keys from DATA_SCHEMA with values provided by the user.
+    """
+
+    hub_address = data[CONF_HOST]
+    websession = async_get_clientsession(hass)
+
+    pv_request = AioRequest(hub_address, loop=hass.loop, websession=websession)
+    hub = Hub(pv_request)
+
+    try:
+        async with async_timeout.timeout(10):
+            await hub.query_user_data()
+    except HUB_EXCEPTIONS:
+        raise CannotConnect
+    if not hub.ip:
+        raise CannotConnect
+
+    # Return info that you want to store in the config entry.
+    return {"title": hub.name}
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Hunter Douglas PowerView."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    def __init__(self):
+        """Initialize the powerview config flow."""
+        self.powerview_config = {}
+
+    async def async_step_user(self, user_input=None):
+        """Handle the initial step."""
+        errors = {}
+        if user_input is not None:
+            if self._host_already_configured(user_input[CONF_HOST]):
+                return self.async_abort(reason="already_configured")
+            try:
+                info = await validate_input(self.hass, user_input)
+
+                return self.async_create_entry(
+                    title=info["title"], data={CONF_HOST: user_input[CONF_HOST]}
+                )
```

Comment:
You're not setting unique ID here. If only homekit has access to the unique ID, consider updating entries in `async_step_homekit` with their correct unique ID. For an example, see how we did it for Tradfri https://github.com/home-assistant/core/pull/34745

Reply:
Copied logic from `tradfri`

Reply:
Nice, updated the existing entry I imported from `yaml`